zerver/views/registration.py

```diff
@@ -529,3 +531,36 @@ def realm_redirect(request: HttpRequest) -> HttpResponse:
         form = RealmRedirectForm()
 
     return render(request, 'zerver/realm_redirect.html', context={'form': form})
+
+@csrf_exempt
+def register_development_user(request: HttpRequest) -> HttpResponseRedirect:
+    realm = get_realm_from_request(request)
+    if realm is None:
+        realm = get_realm('zulip')
+    count = UserProfile.objects.count()
+    name = 'user-%d' % (count,)
+    email = '%s@zulip.com' % (name,)
+    prereg = create_preregistration_user(email, request, realm_creation=False, password_required=False)
+    do_create_user(email=email, password='test', realm=realm,
+                   full_name=name, short_name=name, prereg_user=prereg)
```

Comment:
Should we have this use the same `/accounts/register` call approach used in the register realm code path?  I guess the difference is whether we end up serving a redirect to login as the user afterwards?  

@rishig do you have a preference for whether "create new user" works like "create realm" (in terms of logging you into the app for the user) or just creates the user and doesn't do anything further?

Reply:
Ideally it would log you in (if not hard), since otherwise I think it'd be several clicks to log in.

Reply:
I'm not totally sure of the tradeoffs, but I think having this be as faithful as possible would be nice, since this path is mainly to test the onboarding experience.

Reply:
Yeah, given that context, let's do the same thing we do for realm creation.  Can probably extract a helper function for the `request.POST` manipulation.